homework_1/task_2/progressive_tax/progressiveTax.go

```diff
@@ -0,0 +1,80 @@
+package tax
+
+import (
+	"bufio"
+	"log"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type TaxBracket struct {
+	LowerLimitInclusive float64
+	UpperLimitExclusive float64
+	Percentage float64
+}
+
+func CalculateTax(inputValue float64, taxBrackets []TaxBracket) (float64, error) {
+	var tax float64
+
+	if inputValue < 0 {
+		return 0, errors.New("The input value should be greater than zero.")
+	}
+
+	for index, bracket := range taxBrackets {
+		if inputValue <= 0 {
+			break
+		}
+		if index == len(taxBrackets)-1 {
+			tax += inputValue * bracket.Percentage
+			return tax, nil
+		}
+		tax += math.Min((bracket.UpperLimitExclusive - bracket.LowerLimitInclusive), inputValue) * bracket.Percentage
+		inputValue = inputValue - (bracket.UpperLimitExclusive - bracket.LowerLimitInclusive)
```

Comment:
Reusing the input argument of a function like this can lead to some weird bugs later on and harms readability. I would suggest having another variable `remainingValue` or something similar to use for this.